src/core/server/base.js

```diff
@@ -82,67 +82,37 @@ export default function(routes, createStore) {
-          </Provider>
-        );
-
-        const componentHTML = renderToString(InitialComponent);
-
-        const assets = webpackIsomorphicTools.assets();
-
-        // Get SRI for deployed services only.
-        const sri = isDeployed ? JSON.parse(
-          fs.readFileSync(path.join(config.get('basePath'), 'dist/sri.json'))
-        ) : {};
-
-        const styles = Object.keys(assets.styles).map((style) => {
-          const cssHash = sri[path.basename(assets.styles[style])];
-          if (isDeployed && !cssHash) {
-            throw new Error('Missing SRI Data');
-          }
-          const cssSRI = sri && cssHash ? ` integrity="${cssHash}" crossorigin="anonymous"` : '';
-          return `<link href="${assets.styles[style]}"${cssSRI}
-                        rel="stylesheet" type="text/css" />`;
-        }).join('\n');
-
-        const script = Object.keys(assets.javascript).map((js) => {
-          const jsHash = sri[path.basename(assets.javascript[js])];
-          if (isDeployed && !jsHash) {
-            throw new Error('Missing SRI Data');
-          }
-          const jsSRI = sri && jsHash ? ` integrity="${jsHash}" crossorigin="anonymous"` : '';
-          return `<script src="${assets.javascript[js]}"${jsSRI}></script>`;
-        }).join('\n');
-
-        const HTML = stripIndent`
-          <!DOCTYPE html>
-          <html>
-            <head>
-              <meta charset="utf-8">
-              <title>Isomorphic Redux Demo</title>
-              <meta name="viewport" content="width=device-width, initial-scale=1" />
-              ${styles}
-            </head>
-            <body>
-              <div id="react-view">${componentHTML}</div>
-              <script type="application/json" id="redux-store-state">
-                ${serialize(store.getState())}
-              </script>
-              ${script}
-            </body>
-          </html>`;
-
-        res.header('Content-Type', 'text/html');
-        return res.end(HTML);
-      })
-      .catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(error.stack);
-        res.status(500).end(errorString);
-      });
+      return loadOnServer({...renderProps, store})
+        .then(() => {
+          const InitialComponent = (
+            <Provider store={store} key="provider">
+              <ReduxAsyncConnect {...renderProps} />
+            </Provider>
+          );
+
+          // Get SRI for deployed services only.
+          const sriData = (isDeployed) ? JSON.parse(
+            fs.readFileSync(path.join(config.get('basePath'), 'dist/sri.json'))
+          ) : {};
+
+          const pageProps = {
+            appName: appInstanceName,
+            assets: webpackIsomorphicTools.assets(),
+            component: InitialComponent,
+            head: ReactHelmet.rewind(),
+            env,
+            sriData,
+            store,
+          };
+
+          const HTML = ReactDOM.renderToString(<ServerHtml {...pageProps} />);
```

Comment:
Unfortunately we can do that because Helmet.rewind needs to be called on the server *after* the component is rendered. Using child props caused the data to not be available.